app/backend/src/clj/teet/db_api/core.clj

```diff
@@ -315,6 +316,20 @@
          ~@body))
      (do ~@body)))
 
+(defn- ensure-string-length
+  "Ensure that no strings in tx-data exceed Datomics 4096 character limit."
+  [tx-data]
+  (walk/prewalk
+   (fn [x]
+     (if (and (string? x)
+              (> (.length x) 4096))
```

Comment:
use (count x) for more idiomatic Clojure? Or if there's some compelling reason to use the Java method, a comment explaining it.